app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -27,11 +26,37 @@ class AgendaRepositoryImpl(
     private val db: AgendaDatabase,
 ) : AgendaRepository {
 
-    override suspend fun getAgendaItems(localDate: LocalDate): Flow<List<AgendaItem>> {
-
+    override suspend fun getAgendaItems(localDate: LocalDate) = flow<Flow<List<AgendaItem>>> {
         val startEpochMilli = localDateTimeToZonedEpochMilli(localDate.atStartOfDay())
         val endEpochMilli = localDateTimeToZonedEpochMilli(localDate.atStartOfDay().plusDays(1))
 
+        val reminders = db.getReminderDao().getRemindersOfDate(
+            startEpochMilli = startEpochMilli,
+            endEpochMilli = endEpochMilli
+        ).map {
+            it.map { reminderEntity ->
+                reminderEntity.toReminder()
+            }
+        }
+        val tasks = db.getTaskDao().getTasksOfDate(
+            startEpochMilli = startEpochMilli,
+            endEpochMilli = endEpochMilli
+        ).map {
+            it.map { taskEntity ->
+                taskEntity.toTask()
+            }
+        }
+
+        emit(
+            reminders.combine(tasks) { reminderFlow, taskFlow ->
+                reminderFlow + taskFlow
+            }.map {
+                it.sortedBy { agendaItem ->
+                    agendaItem.startDateAndTime
+                }
+            }
+        )
```

Comment:
In your flow you're emitting flows, instead of just `List<AgendaItem>`. It's comparable to a list of lists and not what you want here.
Instead you should return a `Flow<List<AgendaItem>>` here and in the last line you can use `emitAll(reminders.combine(...) { ... })` which will listen to that flow and send all emissions in the flow you return